src/index.js

```diff
@@ -19,7 +19,9 @@ class ServerlessAppSyncSimulator {
         wsPort: 20003,
         location: '.',
         dynamoDb: {
-          port: get(this.serverless.service, 'custom.dynamodb.start.port', 8000),
+          endpoint: `http://localhost:${get(this.serverless.service, 'custom.dynamodb.start.port', 8000)}`,
+          accessKeyId: 'DEFAULT_ACCESS_KEY',
+          secretAccessKey: 'DEFAULT_SECRET',
```

Comment:
@marcoreni LGTM. Would be great to be able to get credentials from the serverless active profile.

After some investigation, this is doable using
````js
this.serverless.providers.aws.getCredentials()
````

This returns something like:
````json

  credentials: SharedIniFileCredentials {
    expired: false,
    expireTime: null,
    refreshCallbacks: [],
    accessKeyId: 'XxXXXXXxxxxxxXXXX',
    sessionToken: undefined,
    filename: undefined,
    profile: 'xxxxxx',
    disableAssumeRole: false,
    preferStaticCredentials: false,
    tokenCodeFn: [Function],
    httpOptions: null
  },
  region: 'us-east-1'
}
````

which can be passed to the Dynamodb dataloader directly.
However, Amplify's dataloader requires an `endpoint`: https://github.com/aws-amplify/amplify-cli/blob/9c931ed7b5304f475f17db291526a16dcb305699/packages/amplify-appsync-simulator/src/data-loader/dynamo-db/index.ts#L23

Normally, not passing an endpoint, but a region should suffice, but since amplify requires it, this forces us to pass, for example.
`endpoint: 'https://dynamodb.us-east-1.amazonaws.com',` 

That works fine and I can connect to a real DynamoDb. 

I am not sure what to do about this. 

A possibility would be to add an option:
`dynamoDb.useAwsProfile: true`

if `true`, then we use `getCredentials()` to get credentials and region, and we build the endpoint like this
`https://dynamodb.${region}.amazonaws.com`
and pass the credentials.

Additionally, region should be overridable, or taken from the yml in the provider section.

if false, 
we do as it is implemented here. 

All this sounds a bit like overcomplicating things. Since this is supposed to be a mocking tool, I am not even sure this is worth it or the right thing to do. 

What do you think ?

Reply:
I think that we should define the scope of the library. If the primary scope is mocking, then the easier thing to do "end-user" side should be creating and using a local instance. 

This is currently doable by simply configuring `dynamodb-local` plugin without any further configuration.

With this PR, one can also use a custom local endpoint (in my specific usecase, an external docker container), by setting the `endpoint` value inside `appsync-simulator` conf. Since this is, again, a mocked tool, the credentials do not need to be real, but the user has the possibility to set these in case he/she has a custom auth flow for the custom dynamodb instance.

The most "complex" thing would be to connect to a real dynamodb instance.
The old offline-plugin had a similar way of handling the parameters: https://github.com/aheissenberger/serverless-appsync-offline (we're missing the `region` parameter but it's trivial to implement). The user would need to set `endpoint` and `region`, and then set `accessKeyId` and `secretAccessKey` to null so that the standard AWS credentials chain takes control.

I think there's no need to simplify this further, we can write some notes on the readme for this specific use case.

Reply:
@marcoereni I completely agree. Let's keep the scope of this to mocking. 
I get your use-case for using a Dockerized Dynamodb instead of dynamodb-local. Makes sense.
I'll land both PR's during the day.
Thank you again for your feedback and contribution!

Reply:
Great! Just to be on the "safe" side of things I've just added the `region` configuration value.